### ISSUE TITLE
ERROR: client version 1.22 is too old. Minimum supported API version is 1.24, please upgrade your client to a newer version

### ISSUE THREAD

**Issue by nieuwenhuisen** [laradock/laradock#1059]
### Info:
- Docker version (`$ docker --version`): Docker version 17.06.0-ce, build 02c1d87
- Laradock commit (`$ git rev-parse HEAD`): f258b38a18dc915f6eb89ad57dd7bd48f2960a2d
- System info (Mac, PC, Linux): Windows 10

##### What seems to be going wrong?

Execute the command: docker-compose up -d apache2 mysql phpmyadmin

ERROR: client version 1.22 is too old. Minimum supported API version is 1.24, please upgrade your client to a newer version
_____

##### What should be happening instead?

Build and run the containers

**Comment by nieuwenhuisen** [comment]
I have tried to change the version to 2.1 in the docker-compose.yml file, but than I am getting the error "ERROR: plugin not found"

**Comment by nieuwenhuisen** [comment]
It's only when I am using Windows containers. If I switch to Linux containers it's working

**Comment by JacobMillward** [comment]
It seems related to this issue here https://github.com/docker/compose/issues/4106

Seems we need version 2.1 in the docker-compose.yml

**Comment by 0lide** [comment]
Nice, same issue using Windows Containers, but can't use Linux Containers, because i7 and 8gb ram is not enough for docker when try use Linux Containers :D

**Comment by boudewijnk** [comment]
Same issue using Windows Containers on Windows 10.

docker version:
```
Client:
 Version:       17.12.0-ce
 API version:   1.35
 Go version:    go1.9.2
 Git commit:    c97c6d6
 Built: Wed Dec 27 20:05:22 2017
 OS/Arch:       windows/amd64

Server:
 Engine:
  Version:      17.12.0-ce
  API version:  1.35 (minimum version 1.24)
  Go version:   go1.9.2
  Git commit:   c97c6d6
  Built:        Wed Dec 27 20:15:52 2017
  OS/Arch:      windows/amd64
  Experimental: true
```

docker-compose version:
```
docker-compose version 1.18.0, build 8dd22a96
docker-py version: 2.6.1
CPython version: 2.7.14
OpenSSL version: OpenSSL 1.0.2k  26 Jan 2017
```

docker-compose up:
`ERROR: client version 1.22 is too old. Minimum supported API version is 1.24, please upgrade your client to a newer version`

changing version to 2.1 gives:
```
Creating network "xxx" with driver "bridge"
ERROR: plugin not found
```

**Comment by merodiro** [comment]
I have same issue, did anyone found a way to fix it?

**Comment by devmsh** [comment]
Same here!

**Comment by LarryEitel** [comment]
Same issue here. Guess I have to run an earlier version of Windows Docker for now. :(

**Comment by harapeko** [comment]
same!

**Comment by alchimik** [comment]
https://github.com/docker/compose/issues/4106#issuecomment-356443809


>If you are in this (`Windows containers`) mode and try to run/build linux containers you'll probably get errors like:
> | client version 1.22 is too old. Minimum supported API version is 1.24, please up grade your client to a newer version
>or
> | image operating system linux cannot be used
>**Make sure you switch to Linux containers mode if you want to use linux containers.**

**Comment by CondormanFr** [comment]
Same ..

**Comment by muskanaul** [comment]
same issue here, was anyone able to resolve this?

**Comment by gnanasekaranl** [comment]
Same here for windows 10.

**Comment by chidt** [comment]
Me too, It same :(

**Comment by bhx98** [comment]
Me too and with changing docker-compose.yaml 2.0 to 2.4 ,it solved!

**Comment by arliang** [comment]
Same issue solved by updating the [DockerToolbox](https://docs.docker.com/toolbox/toolbox_install_mac/) on Mac

**Comment by MrBandaruk** [comment]
Same issue here. Solved by changing version to `2.1`.

**Comment by 9527andy** [comment]
same issue here, was anyone able to resolve this?
λ docker version                                 
Client:                                          
 Version:           18.06.1-ce                   
 API version:       1.24                         
 Go version:        go1.10.3                     
 Git commit:        e68fc7a                      
 Built:             Tue Aug 21 17:21:34 2018     
 OS/Arch:           windows/amd64                
 Experimental:      false                        
                                                 
Server:                                          
 Engine:                                         
  Version:          18.06.1-ce                   
  API version:      1.38 (minimum version 1.24)  
  Go version:       go1.10.3                     
  Git commit:       e68fc7a                      
  Built:            Tue Aug 21 17:36:40 2018     
  OS/Arch:          windows/amd64                
  Experimental:     false                        

λ docker-compose version
docker-compose version 1.22.0, build f46880fe
docker-py version: 3.4.1
CPython version: 3.6.6
OpenSSL version: OpenSSL 1.0.2o  27 Mar 2018
